app/routes/_layout+/assets.$assetId.duplicate.tsx

```diff
@@ -0,0 +1,177 @@
+import type { ActionArgs, LoaderArgs } from "@remix-run/node";
+import { json, redirect } from "@remix-run/node";
+import {
+  Form,
+  useActionData,
+  useLoaderData,
+  useNavigation,
+} from "@remix-run/react";
+import { parseFormAny, useZorm } from "react-zorm";
+import { z } from "zod";
+import { AssetImage } from "~/components/assets/asset-image";
+import Input from "~/components/forms/input";
+import { Badge, Button } from "~/components/shared";
+import { Spinner } from "~/components/shared/spinner";
+import { db } from "~/database";
+import { duplicateAsset } from "~/modules/asset";
+import { requireAuthSession } from "~/modules/auth";
+import styles from "~/styles/layout/custom-modal.css";
+import {
+  assertIsPost,
+  isFormProcessing,
+  userFriendlyAssetStatus,
+} from "~/utils";
+import { MAX_DUPLICATES_ALLOWED } from "~/utils/constants";
+import { sendNotification } from "~/utils/emitter/send-notification.server";
+import { ShelfStackError } from "~/utils/error";
+
+export const loader = async ({ request, params }: LoaderArgs) => {
+  await requireAuthSession(request);
+  const assetId = params.assetId as string;
+  const asset = await db.asset.findUnique({ where: { id: assetId } });
+  if (!asset) {
+    throw new ShelfStackError({ message: "Asset Not Found", status: 404 });
+  }
+
+  return json({
+    showModal: true,
+    asset,
+  });
+};
+
+const DuplicateAssetSchema = z.object({
```

Comment:
I did some testing and this doesnt work properly because even tho the input is a number it sends a string over the network. What needs to be done is the following:
```ts
const DuplicateAssetSchema = z.object({
  amountOfDuplicates: z.coerce
    .number()
    .min(1, { message: "There should be at least 1 duplicate!" })
    .max(MAX_DUPLICATES_ALLOWED, {
      message: `There can be a max of ${MAX_DUPLICATES_ALLOWED} duplicates!`,
    }),
});

```